### PR TITLE
National Dex: Ban Unobtainables

### PR DESCRIPTION
**Pull request by Spandamn**

### CODE REVIEW

**Comment by Zarel** [comment]
You might as well ban Cosplay Pikachu.

**Comment by Zarel** [comment]
And the LGPE starters.

**Comment by Zarel** [comment]
You remembered Pikachu-Starter but not Eevee-Starter?

**Comment by Spandamn** [comment]
im a bit stupid

**Comment by Zarel** [comment]
Also I would put the banlist inside `onValidateSet` for a clearer message.

File: data/rulesets.js [hunk 1]
```diff
@@ -718,6 +718,11 @@ let BattleFormats = {
 		effectType: 'Rule',
 		name: 'NatDex',
 		onValidateSet(set) {
+			// These Pokemon are still unobtainable
+			let unobtainables = ['Eevee-Starter', 'Floette-Eternal', 'Magearna-Original', 'Pichu-Spiky-eared', 'Pikachu-Belle', 'Pikachu-Cosplay', 'Pikachu-Libre', 'Pikachu-PhD', 'Pikachu-Pop-Star', 'Pikachu-Rock-Star', 'Pikachu-Starter'];
```
Comment (Zarel) [inline data/rulesets.js]:
```suggestion
			const unobtainables = [
				'Eevee-Starter', 'Floette-Eternal', 'Magearna-Original', 'Pichu-Spiky-eared', 'Pikachu-Belle', 'Pikachu-Cosplay', 'Pikachu-Libre', 'Pikachu-PhD', 'Pikachu-Pop-Star', 'Pikachu-Rock-Star', 'Pikachu-Starter',
			];
```

File: data/rulesets.js [hunk 2]
```diff
@@ -718,6 +718,11 @@ let BattleFormats = {
...
+			if (unobtainables.includes(set.species)) {
+				return [`${set.name || set.species} does not exist in Gen ${this.dex.gen}.`];
```
Comment (Zarel) [inline data/rulesets.js]:
```suggestion
				return [`${set.name || set.species} does not exist in Gen 7 or later.`];
```

**Comment by Spandamn** [comment]
Also the snips thing is pretty cool @Zarel

**Pull request merged**